model/k8s/metrics.yaml

```
@@ -126,3 +126,47 @@ groups:
     attributes:
       - ref: network.interface.name
       - ref: network.io.direction
+
+  # k8s.deployment.* metrics
+  - id: metric.k8s.deployment.desired
+    type: metric
+    metric_name: k8s.deployment.desired
+    stability: experimental
+    brief: "Number of desired pods in this deployment"
+    note: |
+      This metric aligns with the `replicas` field of the
+      [K8s Deployment Spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.30/#deploymentspec-v1-apps)
+    instrument: gauge
+    unit: "{pod}"
+  - id: metric.k8s.deployment.available
+    type: metric
+    metric_name: k8s.deployment.available
+    stability: experimental
+    brief: "Total number of available pods (ready for at least minReadySeconds) targeted by this deployment"
+    note: |
+      This metric aligns with the `readyReplicas` field of the
```

```suggestion
      This metric aligns with the `availableReplicas` field of the
```
based on https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8sclusterreceiver/internal/deployment/deployments.go#L26